rust/src/lib.rs

```diff
@@ -259,12 +259,11 @@ pub extern "C" fn zcashlc_init_accounts_table(
                         );
                         CString::new(encoded).unwrap().into_raw()
                     })
-                    .collect();
+                    .collect::<Vec<_>>()
+                    .into_boxed_slice();
                 assert!(v.len() == accounts as usize);
-                unsafe { *capacity_ret.as_mut().unwrap() = v.capacity() };
-                let p = v.as_mut_ptr();
-                std::mem::forget(v);
-                p
+                unsafe { *capacity_ret.as_mut().unwrap() = v.len() };
```

Comment:
A boxed slice has no capacity. This (and the capacity argument) should be removed, and `zcashlc_vec_string_free` should be altered to instead be `zcashlc_boxed_slice_string_free`.